src/Orleans/AssemblyLoader/AssemblyProcessor.cs

```diff
@@ -52,9 +54,10 @@ internal class AssemblyProcessor : IDisposable
         /// <param name="codeGeneratorManager">
         /// The code generator.
         /// </param>
-        public AssemblyProcessor(TypeMetadataCache typeCache, SerializationManager serializationManager, CodeGeneratorManager codeGeneratorManager)
+        /// <param name="loggerFactory"> the logger factory used to create loggers</param>
+        public AssemblyProcessor(TypeMetadataCache typeCache, SerializationManager serializationManager, CodeGeneratorManager codeGeneratorManager, ILoggerFactory loggerFactory)
         {
-            this.logger = LogManager.GetLogger("AssemblyProcessor");
+            this.logger = new LoggerWrapper<AssemblyProcessor>(loggerFactory);
```

Comment:
This API looks clunky. Can we have an extension method on loggerFactory instead? Such as:
this.logger = loggerFactory.GetLegacyLogger<AssemblyProcessor>();

Yo can also have one that accepts as string name instead of a generic T.

Reply:
it has to be a generic T since we want the category to be the class's fullName for logFiltering to work properly. I'd consider an extension method. But I'd rather not using it in our internal code, since that's a lot of changes to this pr. And since this pr is really transitional code anyways. Once the core moved away from Logger. then we won't be using LoggerWrapper in core anymore.

Reply:
I already have one that accepts string as name btw. LoggerWrapper<<'T'>> is built on top of that one LoggerWrapper. But again, I prefer to use oggerWrapper<<'T'>> here

Reply:
sure, it's fine to use the generic version. What I'm saying is have the extension method, since our usage will be only temporal (hopefully), but the API is not, and will live longer in the legacy package. So since the API is clunky for our internal usage, it will be clunky for our customers too.